rayon-core/src/latch.rs

```diff
@@ -189,19 +189,21 @@ impl<'r> Latch for SpinLatch<'r> {
     fn set(&self) {
         let cross_registry;
 
-        let registry = if self.cross {
+        let registry: &Registry = if self.cross {
             // Ensure the registry stays alive while we notify it.
             // Otherwise, it would be possible that we set the spin
             // latch and the other thread sees it and exits, causing
             // the registry to be deallocated, all before we get a
             // chance to invoke `registry.notify_worker_latch_is_set`.
             cross_registry = Arc::clone(self.registry);
-            &cross_registry
+            &*cross_registry
```

Comment:
This is amazing but also terrible as a single character _seems_ to decide if there is a race or not.

Also this isn't criticism on the patch at all but could be a question for the folks at rust-lang if things like that could ever be made more explicit. CC @joshtriplett

Reply:
To be fair, we're also playing with fire to have a `&self` that may not actually live for the whole call. There's a lot about this kind of problem in rust-lang/rust#55005. I have contemplated whether `Latch::set` should use `*const Self` instead, but at some point we'd still be calling atomic `&self` methods with the same core problem.

Reply:
Also, the problem was really on the other branch, borrowing `self.registry`. The `cross_registry` is locally kept alive just fine, and I could have even left this line alone and let deref-coersion convert to `&Registry`, but I chose to make that more explicit.